### PR TITLE
(feat): `{read,write}_dispatched` notebook

### PR DESCRIPTION
**Pull request by ilan-gold**

### CODE REVIEW

**Comment by ivirshup** [comment]
> `python3 -m http.server 8080`

Do you think running this via `subprocess` would be fine? I figure it would keep this more self-contained + runnable on binder.

> update notebook to read from store instead of string so that we can use the consolidated metadata properly

`zarr.open_consolidated` instead of `zarr.open`?

**Comment by ilan-gold** [comment]
I think the thing about `subprocess` is that I don't think you'll be able to monitor what's going in and out.  We could route the output to another file but I'll to fiddle around with it.

File: requirements.txt [hunk 1]
```diff
@@ -3,3 +3,5 @@ sphinx>=1.4
 ipykernel
 nbsphinx>=0.4
 nbclean
+requests
+aiohttp
```
Comment (flying-sheep) [inline requirements.txt:7]: I don’t see those being used anywhere in the code.

If you’re using some optional feature that depends on those, specify that optional feature instead: `somepackage[web]` or so.

**Review by flying-sheep — CHANGES_REQUESTED** [review]

**Comment by ivirshup** [comment]
> I think the thing about subprocess is that I don't think you'll be able to monitor what's going in and out. 

IDK that it's critical to monitor this.

But this should be possible with something like:

```python
from subprocess import Popen, PIPE

proc = Popen(["python3", "-m", "http.server", "8080"], stdout=PIPE, stderr=PIPE)
```

File: requirements.txt (hunk 1 above)
Comment (ivirshup) [reply requirements.txt:7]: :+1:

**Comment by ivirshup** [comment]
@ilan-gold, also any file that's new enough for read_dispatched to work on wouldn't need backwards compat.

File: requirements.txt (hunk 1 above)
Comment (ilan-gold) [reply requirements.txt:7]: Ok can do.

**Comment by flying-sheep** [comment]
`subprocess.run` is the preferred interface

**Comment by ivirshup** [comment]
@flying-sheep I believe `run` will wait until the process closes:

image

but we want something that continues to run on another thread.

Looking into it a bit more, I think this would work (but not on windows):

```python
import os
from subprocess import Popen, PIPE

proc = Popen(["python3", "-m", "http.server", "8080"], stderr=PIPE)
os.set_blocking(proc.stderr.fileno(), False)

# Run this each time you want to see the server output
for line in proc.stderr.readlines():
    print(line.decode("utf-8").strip())
```

**Comment by ilan-gold** [comment]
I guess people don't necessarily have to see what is going on, but I also think it's a nice thing.  Otherwise, there isn't much ostensibly making this different than either the dask demo or a simpler `read_dispatched` that just downloads the data first.  But maybe people are already familiar enough with access patterns that they'll understand what is going on here and why it is significant.

**Comment by ilan-gold** [comment]
So if we want, we can advise that they do the above on non-windows and then just use a shell in windows,

**Comment by ilan-gold** [comment]
It seems like piping is not working as expected for me.

**Comment by ivirshup** [comment]
Yeah, I also gave it a shot and was running into dask blocking during read... For some reason it worked if I just used `sc.datasets.pbmc3k_processed().raw.to_adata()`?

But if this only happens when the server is run from the notebook, I assume it's some concurrency issue, and it might be difficult to figure out if the problem is jupyter, dask, or `os.set_blocking`

----

While I would really like to be able to run this whole notebook without having to run something from a separate terminal, it's not required.

I would also note that there is just a ton of output from the initial read, even with consolidated metadata (since all the other values need to load). This might be a little hard to parse.

**Comment by ilan-gold** [comment]
@ivirshup That is true.  I also noticed that.  Not sure what to do about that.  Maybe it makes it not worth it to use this, but I do think that since the whole point of this is "how can we do certain operations well?" it may be essential.

**Comment by ivirshup** [comment]
I'm going to merge since this is one of the last things blocking a release candidate. We can clean it up further separately.
